nexus/social_media/services.py

```diff
@@ -0,0 +1,166 @@
+# Third Party Stuff
+import requests
+from django.conf import settings
+from django.utils import timezone
+from rest_framework import status
+
+# nexus Stuff
+from nexus.base import exceptions as exc
+from nexus.social_media.models import Post
+
+
+def update_post_object(post):
+    """Update post instance after the post is published.
+    :param post: Post model instance.
+    """
+    post.posted_time = timezone.now()
+    post.is_posted = True
+    post.save()
+
+
+def upload_image_to_linkedin(author, headers, post, api_url_base, linkedin):
+    """Upload image to linkedin for given post.
+    :param author: Owner of the post to be used for response POST data.
+    :param headers: HTTP headers required for successful HTTP request.
+    :param post: Post model instance.
+    :param api_url_base: Base url for LinkedIn API.
+    :param linkedin: LINKEDIN_AUTH object taken from settings.
+    :returns: A valid HTTP response.
+    """
+    post_data_assets = {
+        "registerUploadRequest": {
+            "recipes": [
+                "urn:li:digitalmediaRecipe:feedshare-image"
+            ],
+            "owner": author,
+            "serviceRelationships": [
+                {
+                    "relationshipType": "OWNER",
+                    "identifier": "urn:li:userGeneratedContent"
+                }
+            ]
+        }
+    }
+
+    api_url_assets = f"{api_url_base}assets?action=registerUpload"
+
+    response_assets = requests.post(api_url_assets, json=post_data_assets,
+                                    headers=headers)
+    if response_assets.status_code == status.HTTP_200_OK:
+        json_response_assets = response_assets.json()
+        uploadUrl = json_response_assets.get('value')\
+                                        .get('uploadMechanism')\
+                                        .get('com.linkedin.digitalmedia.uploading.MediaUploadHttpRequest')\
+                                        .get('uploadUrl')
+
+        asset = json_response_assets.get('value').get('asset')
+        with open(post.image.file.name, 'rb') as f:
+            response_upload_image = requests.post(uploadUrl, data=f.read(),
```

Comment:
Just to confirm, isn't that `f.read()` will be equivalent to `post.image`?

Reply:
No. `post.image` gives the path of the image w.r.t the `social_media` directory. An example of such path would be `social_media/post/9rCe0BL7S5CHlqPcuATY4g.png`

Reply:
What about post.image.file?

Reply:
`post.image.file` and `post.image.file.name` gives the full name of the file wrt to the home of your system.

Reply:
If you wish you can do:
```suggestion
            response_upload_image = requests.post(uploadUrl, data=post.image.file.read(),
```
and remove the with statement